net/bridge.go

```diff
@@ -327,22 +328,24 @@ func (b bridgeImpl) initPrep(config *BridgeConfig) error {
 	// fails. Bridges take the lowest MTU of their interfaces. So
 	// instead we create a temporary interface with the desired MTU,
 	// attach that to the bridge, and then remove it again.
-	dummy := &netlink.Dummy{LinkAttrs: netlink.NewLinkAttrs()}
-	dummy.LinkAttrs.Name = "vethwedu"
-	if err = netlink.LinkAdd(dummy); err != nil {
+	var dummy netlink.Link
+	dummy = &netlink.Dummy{LinkAttrs: netlink.NewLinkAttrs()}
+	dummy.Attrs().Name = WeaveDummyIfName
+	if err = LinkAddIfNotExist(dummy); err != nil {
 		return errors.Wrap(err, "creating dummy interface")
 	}
+	defer func() {
+		if err = netlink.LinkDel(dummy); err != nil {
+			err = errors.Wrap(err, "deleting dummy interface")
```

Comment:
I think assigning to the local variable `err` at this point has no effect.
If `err` is made a named return value from `initPrep` it would work, but suppose both `LinkSetMTU` and `LinkDel` fail, we'd only see one error.
Maybe use a different `err` to record the `LinkDel()` result and use `Wrap()` to add the previous error (if any).

Reply:
Hmm, `Wrap()` can only wrap one error.  Maybe just add the text of this one to the previous one, if it exists.